aten/src/ATen/NestedTensorImpl.h

```diff
@@ -142,6 +155,18 @@ struct TORCH_API NestedTensorImpl : public c10::TensorImpl {
   c10::intrusive_ptr<TensorImpl> shallow_copy_and_detach_core(
       VariableVersion&& version_counter,
       bool allow_tensor_metadata_change) const;
+
+  inline c10::DispatchKeySet generate_buffer_key_set() const {
+    const auto backend = this->key_set_.highestBackendKey();
+    auto buffer_key_set = c10::DispatchKeySet{c10::DispatchKey::Dense} |
+        c10::DispatchKeySet{backend};
+    const bool Autograd = this->key_set_.has_any(c10::autograd_dispatch_keyset);
+
+    buffer_key_set = Autograd
+        ? getAutogradRelatedKeySetFromBackend(backend) | buffer_key_set
+        : buffer_key_set;
+    return buffer_key_set;
+  }
```

Comment:
It's likely I'm misunderstanding something here, but I thought that we don't want to track autograd stuff for the buffer. Why are autograd-related keys added to the buffer_key_set?

Reply:
yeah this was my russian doll comment. and has to do with unbind also being a view and the implementation. An alternative would be  get_buffer and a get_buffer_view and have separate key gens for these two. This is the more heterogenous solution and I think more true to the spirit of what is going on when we call get_buffer().

Reply:
I'll also add that we should run Tensor operations on the buffer within a `no_grad()` context (and I think we do in general).

Reply:
@alban The interface between NT->T is showing up here: https://github.com/pytorch/pytorch/blob/cd18b78daa25972efb39077619cfc86a886ff4fa/aten/src/ATen/native/nested/NestedTensorMath.cpp#L118

For functional ops the flow is variable_type_everything which registers the appropriate grad_fun and then redispatches after grad_keys. This is where nested_tensor kernels that call get_buffer() are all simply implementation details. But for view ops that use the buffer, like unbind linked above, this constraint is no longer the case.

Reply:
I'll note that this PR does not attempt to fix nestedtensor_unbind's improper use of get_buffer(). The unbind function is creating a Tensor view of a Nested Tensor. The work to handle this scenario is done in a follow up PR based on this branch: #83371